### DISCUSSION TITLE
Open local desktop files

### DISCUSSION THREAD

**Discussion by mkant078** [cefsharp/CefSharp#3352] [Q&A]
Hi,
I am new to GitHub, so please do not shoot me if I have a stupid question. 
For our intranet I need to open local desktop applications (MS Office Excel) by clicking a link on the intranet webpage. Since all browsers have a level of security that make this impossible, I thought CefSharp could be a good solution to make my own browser without this security. Or at least to turn off this security setting. I think it has something to do with a Local Storage setting, correct me if I am wrong. 
Is this possible with CefSharp? If yes, can someone point me in the right direction. 
Thanks

**Comment by amaitland** [comment]
> Since all browsers have a level of security that make this impossible

That's not exactly correct,  every major browser supports the opening of external applications on Windows. See https://nishantrana.me/2020/09/22/use-custom-browser-protocol-to-launch-desktop-applications-from-dynamics-365/

I'd suggest evaluating this option before writing a custom browser. 

If you decide to use CefSharp see https://github.com/cefsharp/CefSharp/wiki/General-Usage#protocol-execution for an example that you should be able to adapt.
